.buildkite/pipeline.yml

```diff
@@ -570,25 +570,34 @@
       --test_tag_filters=tests_dir_R,-flaky,-multi_gpu --test_env=RAY_USE_MULTIPROCESSING_CPU_COUNT=1
       rllib/...
 
-- label: ":octopus: Tune tests and examples {1/2}"
+- label: ":octopus: Tune tests and examples {1/2; no RLlib}"
   conditions: ["RAY_CI_TUNE_AFFECTED"]
   commands:
     - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/travis/upload_build_info.sh; fi }; trap cleanup EXIT
     - TUNE_TESTING=1 PYTHON=3.7 ./ci/travis/install-dependencies.sh
     # Because Python version changed, we need to re-install Ray here
     - rm -rf ./python/ray/thirdparty_files; rm -rf ./python/ray/pickle5_files; ./ci/travis/ci.sh build
-    - bazel test --config=ci $(./scripts/bazel_export_options) --test_tag_filters=-example,-flaky,-py37,-soft_imports,-gpu_only python/ray/tune/...
-    - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=example,-tf,-pytorch,-py37,-flaky,-soft_imports,-gpu_only python/ray/tune/...
+    - bazel test --config=ci $(./scripts/bazel_export_options) --test_tag_filters=-example,-flaky,-py37,-soft_imports,-gpu_only,-rllib python/ray/tune/...
+    - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=example,-tf,-pytorch,-py37,-flaky,-soft_imports,-gpu_only,-rllib python/ray/tune/...
 
-- label: ":octopus: Tune tests and examples {2/2}"
+- label: ":octopus: Tune tests and examples {2/2; no RLlib}"
   conditions: ["RAY_CI_TUNE_AFFECTED"]
   commands:
     - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/travis/upload_build_info.sh; fi }; trap cleanup EXIT
     - TUNE_TESTING=1 PYTHON=3.7 ./ci/travis/install-dependencies.sh
     # Because Python version changed, we need to re-install Ray here
     - rm -rf ./python/ray/thirdparty_files; rm -rf ./python/ray/pickle5_files; ./ci/travis/ci.sh build
-    - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=tf,-pytorch,-py37,-flaky,-soft_imports,-gpu_only python/ray/tune/...
-    - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=-tf,pytorch,-py37,-flaky,-soft_imports,-gpu_only python/ray/tune/...
+    - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=tf,-pytorch,-py37,-flaky,-soft_imports,-gpu_only,-rllib python/ray/tune/...
+    - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=-tf,pytorch,-py37,-flaky,-soft_imports,-gpu_only,-rllib python/ray/tune/...
+
+- label: ":octopus: Tune tests and examples {using RLlib}"
+  conditions: ["RAY_CI_TUNE_AFFECTED"]
```

Comment:
should you add RAY_CI_RLLIB_AFECTED here, so that these will run whenever rllib is changed?

Reply:
+1, it may also be worthwhile to take another pass over the condition definitions in [determine_tests_to_run.py](https://sourcegraph.com/github.com/ray-project/ray/-/blob/ci/travis/determine_tests_to_run.py?L77-78) to double check we're not over/under-testing. Interestingly enough I stumbled upon a [recent PR](https://github.com/ray-project/ray/pull/19656) that removes an unused `RAY_CI_ONLY_RLLIB_AFFECTED` condition, not sure if we might actually want to add it back and use it for other RLLib test suites.

Reply:
Great point!
I re-instated the RLLIB_AFFECTED vs RLLIB_DIRECTLY_AFFECTED differentiation.

`RLLIB_AFFECTED` runs a sub-set of RLlib tests (the high-level learning tests + examples) and is triggered by changes in tune, etc.., non-RLlib sub-folders.

`RLLIB_DIRECTLY_AFFECTED` runs all RLlib tests and is only triggered upon changes in the rllib source files